### ISSUE TITLE
support #[serde(skip_serializing_if = "path")]

### ISSUE THREAD

**Issue by RagibHasin** [GREsau/schemars#26]
Currently this causes an error like

mismatched types
expected reference `&std::option::Option<_>`
   found reference `&<task::Task as schemars::JsonSchema>::json_schema::_SchemarsDefaultSerialize<std::option::Option<std::time::Duration>>`

I think it should mark that field as not required.

Thanks in advance.

**Comment by RagibHasin** [comment]
My the field causing error looks like,

```rust
    #[schemars(with = "Option<String>")]
    #[serde(
        default,
        with = "duration_utils::opt",
        skip_serializing_if = "Option::is_none"
    )]
    pub duration: Option<StdDuration>,
```

removing any of the serde attributes makes it OK. I don't get what is wrong.

**Comment by GREsau** [comment]
Yep that's definitely a bug in schemars, I'll try to get a fix out asap

**Comment by GREsau** [comment]
This should be fixed in v0.7.1 which is now on crates.io: https://crates.io/crates/schemars/0.7.1

@RagibHasin could you please re-test and let me know if it works for you?

**Comment by RagibHasin** [comment]
Thanks for that quick a reply! It solved the problem. I was taking a break for a while. So, sorry for my being late.
